Milestone 1 : Prepare data architecture documentation

kiptubei opened the pull request:
Created ERD diagram for Posts, Coments, Users ,Frienships and Reactions
Added Gravatar gem and dependencies to User Model

kiptubei committed:
Created ERD diagram, added gravatar dependencies

davidmrtz-dev reviewed (changes requested):
## Hi Team.  👨‍🚀 
### Very good job so far.

However, there are some issues that need to be addressed before moving on.  👨‍💻  ⬇️ 
Please check out below.

****
- [ ] User <-> Posts Relation.
Users can have multiple posts, and multiple posts can belong to one user.
Currently, we have a relationship that specifies a user can only have one post and a post can only belong to one user.

![Screen Shot 2020-08-25 at 9 14 18 PM](https://user-images.githubusercontent.com/52430951/91247474-fb069c80-e717-11ea-9d5c-ebf8d5069605.png)

Please fix this, you can follow the same structure as you have un your User <-> Friendship relation:

![Screen Shot 2020-08-25 at 9 16 06 PM](https://user-images.githubusercontent.com/52430951/91247584-3a34ed80-e718-11ea-9d86-66251ef4acd1.png)
****

- [ ] User <-> Friendship Relation.
Currently, we have this in the relation:
![Screen Shot 2020-08-25 at 9 19 09 PM](https://user-images.githubusercontent.com/52430951/91247763-a6175600-e718-11ea-89ac-a98ca1ccace3.png)

In order to find all friendship we needs friendships with status == ' confirmed' and with (user_id == current_user or friend_id == current_user(

Consider the following example:

![Screen Shot 2020-08-25 at 9 21 29 PM](https://user-images.githubusercontent.com/52430951/91247886-f68eb380-e718-11ea-9e1f-40017ebc3fbe.png)

****

- [ ] User table has image_link column ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone1/imglink.png)  from  [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))


****

### **Please make the changes required and submit for re-review once you are ready!** 

****

### Status: Changes required. ♻️  
****

Happy coding!  💻  
Have a great day 🙂

If you need more clarifications kindly reach me on slack.
**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

kiptubei committed:
Added one to many relationship between users and posts, Added one to any user and freindship relation ship,(user can invite many freinds), added a many friend_id to user relationship (a user can be invited by many people to be friends,i.e they are in the friend id column)

kiptubei commented:
Added one to many relationships between users and posts, 
Added one to any user and friendship relationship,(user can invite many friends),
Added a many friend_id to user relationship (a user can be invited by many people to be friends,i.e they are in the friend id column)"

wathigo reviewed (changes requested):
## Changes Required.
Great work implementing the suggested changes.

### REQUIRED
**Friendship feature**
* Let's fix the relationship between friend and friendship record. 
The friendship feature association can be summarized as:
User -> Friendship = one -> many
Friend -> Friendship = one -> many.
This means a user can be referenced in many friendships. ie A user can send many friend requests and also a friend(a user receiving a friend request) can be referenced in many friendships. ie receive many friend requests.
Let change the relationship between Friend(the user referenced using `friend_id` and friendship to one -> many. Currently, the relationship is many -> one.

**User -> Reaction**
* The relationship between user and relation should be one -> many.
A user can react to many comments/posts while a single reaction can only belong to one user.
The current association is one -> one.

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo.**

kiptubei committed:
Added friend to friendship relationship one-to-many (friend_id references user_id, a user can recieve many friend requests),Changed user comments relationship to one-to-many

kiptubei commented:
Added friend to friendship relationship one-to-many (friend_id references user_id, a user can receive many friend requests),
Changed user comments relationship to one-to-many

mikenath223 reviewed (changes requested):
### Hi Team 👋

Nice work on this so far.

Just a few more fixes left:

- [ ] Please remove the stickler.yml file in the base directory. It isn't necessary.

- [ ] You shouldn't invert the friendship association on itself. Rather link it twice to the user table to indicate that it's the joins table for the user associations [Screenshot](https://user-images.githubusercontent.com/52430951/91247886-f68eb380-e718-11ea-9e1f-40017ebc3fbe.png).


Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

kiptubei committed:
Removed stickler.yml from base directory, linked twice the user table to freindship table to indicate that it's the joins table for the user associations

kiptubei commented:
Removed stickler.yml from base directory,
Linked the user table to friendship table twice ,to indicate that it's the joins table for the user associations

mikenath223 reviewed (approved):
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻
